File: ocaml/ex2_1_2.ml
class person = object
  method send_email (content: string): unit = ()
end

let send_new_year_email (dst: < send_email: string -> unit; .. >): unit =
  dst#send_email ""

let _ =
  let pr: person = new person in
  send_new_year_email pr

File: scala/ex2_1_2.scala
import reflect.Selectable.reflectiveSelectable

class Person(val email: String)
class Student(val email: String, val grade: Int)

def sendEmail(person: { val email: String }): Unit =
  val email: String = person.email

@main def main =
  val st: Student = Student("", 0)
  sendEmail(st)
